### PR TITLE
View counter middle ware

### PR DESCRIPTION
**Pull request by charliew-rsi**
Added code that counts views

### CODE REVIEW

File: src/components/ArticlePreview.js [hunk 1]
```diff
@@ -21,56 +21,66 @@ const mapDispatchToProps = dispatch => ({
...
+      props.unfavorite(slug);
     } else {
-      props.favorite(article.slug);
+      props.favorite(slug);
     }
   };
 
+  const onHandleLinkClick = (event, slug) => {
+    event.persist();
+    console.log(slug);
```
Comment (wesdollar) [inline src/components/ArticlePreview.js]: Remove `console.log`

File: src/components/ArticlePreview.js [hunk 2]
```diff
@@ -21,56 +21,66 @@ const mapDispatchToProps = dispatch => ({
...
+    if (favorited) {
+      props.unfavorite(slug);
     } else {
-      props.favorite(article.slug);
+      props.favorite(slug);
     }
   };
 
+  const onHandleLinkClick = (event, slug) => {
+    event.persist();
```
Comment (wesdollar) [inline src/components/ArticlePreview.js]: Why did you use `event.persist()` here?

File: src/components/ArticlePreview.js [hunk 3]
```diff
@@ -21,56 +21,66 @@ const mapDispatchToProps = dispatch => ({
...
+    addView(slug);
+  };
+
   return (
     <div className="article-preview">
       <div className="article-meta">
         <div className="info">
-          <Link className="author" to={`/@${article.author.username}`}>
-            {article.author.username}
+          <Link className="author" to={`/@${username}`}>
```
Comment (wesdollar) [inline src/components/ArticlePreview.js]: I don't see where `username` is defined

File: src/components/ArticlePreview.js [hunk 4]
```diff
@@ -21,56 +21,66 @@ const mapDispatchToProps = dispatch => ({
...
+  } = props.article;
+
+  const favoriteButtonClass = favorited ? FAVORITED_CLASS : NOT_FAVORITED_CLASS;
 
   const handleClick = ev => {
     ev.preventDefault();
-    if (article.favorited) {
-      props.unfavorite(article.slug);
+    if (favorited) {
+      props.unfavorite(slug);
```
Comment (wesdollar) [inline src/components/ArticlePreview.js]: Destructure this to remain consistent with "code standards"

File: src/components/ArticlePreview.js [hunk 5]
```diff
@@ -21,56 +21,66 @@ const mapDispatchToProps = dispatch => ({
...
     } else {
-      props.favorite(article.slug);
+      props.favorite(slug);
```
Comment (wesdollar) [inline src/components/ArticlePreview.js]: Destructure

File: src/components/ArticlePreview.js (hunk 1 above)
Comment (charliew-rsi) [reply src/components/ArticlePreview.js]: removing

File: src/components/ArticlePreview.js (hunk 2 above)
Comment (charliew-rsi) [reply src/components/ArticlePreview.js]: event.persist() appears to be redundant, as well as passing slug as a parameter. This function needs no knowledge of the event and already has access to the slug variable. The event.persist() would be necessary if we needed the knowledge of the event even after the link routes to the new page Removing redundant parameters.

File: src/components/ArticlePreview.js (hunk 3 above)
Comment (charliew-rsi) [reply src/components/ArticlePreview.js]: username is defined by destructuring on line 32. Prior testing confirmed that none of these values reported undefined.

File: src/components/ArticlePreview.js (hunk 4 above)
Comment (charliew-rsi) [reply src/components/ArticlePreview.js]: Adding functions to destructuring process

File: src/components/ArticlePreview.js (hunk 5 above)
Comment (charliew-rsi) [reply src/components/ArticlePreview.js]: Same

File: src/components/ArticlePreview.js (hunk 3 above)
Comment (wesdollar) [reply src/components/ArticlePreview.js]: Ok, didn't see it. Thanks!

File: src/services/article.js [hunk 6]
```diff
@@ -127,3 +127,8 @@ export const fetchAllTags = () => {
     }
   );
 };
+
+export const addView = articleSlug => {
+  console.log(articleSlug);
```
Comment (wesdollar) [inline src/services/article.js]: `console.log`... self-review, please.

**Pull request merged**